GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+            //*********************************************************************************************
+            Group newGroup = new Group
+            {
+                DisplayName = "newGroup" + Helper.GetRandomString(8),
+                Description = "Best Group ever",
+                MailNickname = "group" + Helper.GetRandomString(4),
+                MailEnabled = false,
+                SecurityEnabled = true
+            };
+            try
+            {
+                await client.Groups.AddGroupAsync(newGroup);
+                Console.WriteLine("\nNew Group {0} was created", newGroup.DisplayName);
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError creating new Group {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+            return newGroup;
+
+            #endregion
+        }
+
+        private static async Task PrintAllRoles(IActiveDirectoryClient client, string searchString)
+        {
+            #region Get All Roles
+
+            //*********************************************************************
+            // Get All Roles
+            //*********************************************************************
+            List<IDirectoryRole> foundRoles = null;
+            try
+            {
+                IPagedCollection<IDirectoryRole> rolesCollection = await client.DirectoryRoles.ExecuteAsync();
+                foundRoles = rolesCollection.CurrentPage.ToList();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Roles {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            if (foundRoles != null && foundRoles.Count > 0)
+            {
+                foreach (IDirectoryRole role in foundRoles)
+                {
+                    Console.WriteLine("\n Found Role: {0} {1} {2} ",
+                        role.DisplayName, role.Description, role.ObjectId);
+                }
+            }
+            else
+            {
+                Console.WriteLine("Role Not Found {0}", searchString);
+            }
+
+            #endregion
+        }
+
+        private static async Task PrintServicePrincipals(IActiveDirectoryClient client)
+        {
+            #region Get Service Principals
+
+            //*********************************************************************
+            // get the Service Principals
+            //*********************************************************************
+            IPagedCollection<IServicePrincipal> servicePrincipals = null;
+            try
+            {
+                servicePrincipals = await client.ServicePrincipals.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Program.WriteError("\nError getting Service Principal {0} {1}",
```

Comment:
>WriteError [](start = 24, length = 10)

How about sending exception to WriteError. That way you wont be repeating the following check in every place.
e.InnerException != null ? e.InnerException.Message : ""